scripts/templates/helper.py

```diff
@@ -1013,7 +1013,39 @@ def make_pfncb_param_type(namespace, tags, obj):
 
 """
 Public:
-    returns a dict of auto-generated c++ parameter validation checks
+    returns an appropriate bounds_check call based on a bounds_check tag from
+    the spec
+"""
+def get_bounds_check(param_dict, bounds_error):
+    bounds_call = re.search(r'\[(bounds\(.*\))\]', param_dict['desc']).group(1)
+    bounds_call = re.sub(
+        r'bounds\(',
+        'auto {0} = bounds({1}, '.format(bounds_error, param_dict['name']),
+        bounds_call,
+    )
+    bounds_call += '; {0} != UR_RESULT_SUCCESS'.format(bounds_error)
+    # Image needs its own helper, since function signature wise it's identical
+    # to buffer rect
+    if'image' in param_dict['name'].lower():
+        return re.sub(r'bounds\(', 'boundsImage(', bounds_call)
+
+    # USM bounds checks need the queue handle parameter to be able to use the
+    # GetMemAllocInfo entry point
+    if'void' in param_dict['type']:
+        # TODO: this could be more complex, we could pull the full function
```

Comment:
makes sense, but, if possible, it would be nice to have a check somewhere (maybe during param validation) with that produces a nice error message if bounds is used where `hQueue` isn't defined.